platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpoint.java

```diff
@@ -159,73 +150,64 @@ public Response create(MultipartBody multipartBody, @Context UriInfo requestUriI
    * @param attachment the attachment to copy and extract.
    * @return The file of the copied attachment.
    */
+  @Nullable
   private File createFileFromAttachement(Attachment attachment) {
     InputStream inputStream = null;
-    String filename = null;
-    File newFile = null;
-    if (attachment.getContentDisposition() != null) {
-      filename =
-          attachment
-              .getContentDisposition()
-              .getParameter(FILENAME_CONTENT_DISPOSITION_PARAMETER_NAME);
-    }
-
-    if (StringUtils.isEmpty(filename)) {
-      LOGGER.debug("Filename not found, using default.");
-      filename = DEFAULT_FILE_NAME;
-    } else {
-      filename = FilenameUtils.getName(filename);
-      LOGGER.debug("Filename: {}", filename);
-    }
-
     try {
       inputStream = attachment.getDataHandler().getInputStream();
-      if (inputStream != null && inputStream.available() == 0) {
+      if (inputStream == null) {
+        LOGGER.debug("No file attachment found");
+        return null;
+      }
+
+      if (inputStream.available() == 0) {
         inputStream.reset();
       }
+
+      String filename = getFileName(attachment.getContentDisposition());
+
+      if (!filename.endsWith(JAR_EXT) && !filename.endsWith(KAR_EXT)) {
+        LOGGER.debug("Wrong file type: {}", FilenameUtils.getExtension(filename));
+        return null;
+      }
+
+      File uploadDir = new File(defaultFileLocation);
+      if (!uploadDir.exists() && !uploadDir.mkdirs()) {
+        LOGGER.info("Unable to make directory {}", uploadDir.getAbsolutePath());
+      }
+
+      File newFile = new File(uploadDir, filename);
```

Comment:
❓ If you couldn't make the `uploadDir` above...

Reply:
Yep. You are absolutely right. This is precisely how it was before and I didn't feel like rewriting tests since the healthy majority of this service will change in the near term.